RecoTracker/MkFitCore/interface/Hit.h

```diff
@@ -0,0 +1,329 @@
+  }
+
+  inline float getEtaErr2(float x, float y, float z, float exx, float eyy, float ezz, float exy, float exz, float eyz) {
+    const float rad2 = getRad2(x, y);
+    const float detadx = -x / (rad2 * std::sqrt(1 + rad2 / (z * z)));
+    const float detady = -y / (rad2 * std::sqrt(1 + rad2 / (z * z)));
+    const float detadz = 1.0f / (z * std::sqrt(1 + rad2 / (z * z)));
+    return detadx * detadx * exx + detady * detady * eyy + detadz * detadz * ezz + 2.0f * detadx * detady * exy +
+           2.0f * detadx * detadz * exz + 2.0f * detady * detadz * eyz;
+  }
+
+  inline float getPxPxErr2(float ipt, float phi, float vipt, float vphi) {  // ipt = 1/pT, v = variance
+    const float iipt2 = 1.0f / (ipt * ipt);                                 //iipt = 1/(1/pT) = pT
+    const float cosP = std::cos(phi);
+    const float sinP = std::sin(phi);
+    return iipt2 * (iipt2 * cosP * cosP * vipt + sinP * sinP * vphi);
+  }
+
+  inline float getPyPyErr2(float ipt, float phi, float vipt, float vphi) {  // ipt = 1/pT, v = variance
+    const float iipt2 = 1.0f / (ipt * ipt);                                 //iipt = 1/(1/pT) = pT
+    const float cosP = std::cos(phi);
+    const float sinP = std::sin(phi);
+    return iipt2 * (iipt2 * sinP * sinP * vipt + cosP * cosP * vphi);
+  }
+
+  inline float getPzPzErr2(float ipt, float theta, float vipt, float vtheta) {  // ipt = 1/pT, v = variance
+    const float iipt2 = 1.0f / (ipt * ipt);                                     //iipt = 1/(1/pT) = pT
+    const float cotT = 1.0f / std::tan(theta);
+    const float cscT = 1.0f / std::sin(theta);
+    return iipt2 * (iipt2 * cotT * cotT * vipt + cscT * cscT * cscT * cscT * vtheta);
+  }
+
+  struct MCHitInfo {
+    MCHitInfo() {}
+    MCHitInfo(int track, int layer, int ithlayerhit, int mcHitID)
+        : mcTrackID_(track), layer_(layer), ithLayerHit_(ithlayerhit), mcHitID_(mcHitID) {}
+
+    int mcTrackID_;
+    int layer_;
+    int ithLayerHit_;
+    int mcHitID_;
+
+    int mcTrackID() const { return mcTrackID_; }
+    int layer() const { return layer_; }
+    int mcHitID() const { return mcHitID_; }
+    static void reset();
+  };
+  typedef std::vector<MCHitInfo> MCHitInfoVec;
+
+  struct MeasurementState {
+  public:
+    MeasurementState() {}
+    MeasurementState(const SVector3& p, const SVector6& e) : pos_(p), err_(e) {}
+    MeasurementState(const SVector3& p, const SMatrixSym33& e) : pos_(p) {
+      for (int i = 0; i < 6; ++i)
+        err_[i] = e.Array()[i];
+    }
+    const SVector3& parameters() const { return pos_; }
+    SMatrixSym33 errors() const {
+      SMatrixSym33 result;
+      for (int i = 0; i < 6; ++i)
+        result.Array()[i] = err_[i];
+      return result;
+    }
+    SVector3 pos_;
+    SVector6 err_;
+  };
+
+  class Hit {
+  public:
+    Hit() : mcHitID_(-1) {}
+
+    Hit(const SVector3& position, const SMatrixSym33& error, int mcHitID = -1)
+        : state_(position, error), mcHitID_(mcHitID) {}
+
+    ~Hit() {}
```

Comment:
any reason to declare this explicitly? or maybe better use the default?

Reply:
Addressed in Round 1 follow-up PR trackreco/cmssw#75 commit: https://github.com/trackreco/cmssw/pull/75/commits/2fe41d883b291d06218f31566de08530396bf413